python/seldon_core/metrics.py

```diff
@@ -1,9 +1,168 @@
+from prometheus_client.core import (
+    HistogramMetricFamily,
+    GaugeMetricFamily,
+    CounterMetricFamily,
+    CollectorRegistry,
+)
+from prometheus_client import exposition
+from prometheus_client.utils import floatToGoString
+
+from multiprocessing import Manager
+
+import numpy as np
+
 from typing import List, Dict
+import logging
+import json
+import os
+
+
+logger = logging.getLogger(__name__)
+
+NONIMPLEMENTED_MSG = "NOT_IMPLEMENTED"
+
+ENV_SELDON_DEPLOYMENT_NAME = "SELDON_DEPLOYMENT_ID"
+ENV_MODEL_NAME = "PREDICTIVE_UNIT_ID"
+ENV_MODEL_IMAGE = "PREDICTIVE_UNIT_IMAGE"
+ENV_PREDICTOR_NAME = "PREDICTOR_ID"
+ENV_PREDICTOR_LABELS = "PREDICTOR_LABELS"
 
 COUNTER = "COUNTER"
 GAUGE = "GAUGE"
 TIMER = "TIMER"
 
+# This sets the bins spread logarithmically between 0.001 and 30
+BINS = [0] + list(np.logspace(-3, np.log10(30), 50)) + [np.inf]
```

Comment:
any plan to make this BIN size customizable by users ?

Reply:
That's a great idea! Would you mind opening an issue for that?
This PR is more meant to make sure that metrics works in (almost) same way with `executor` as it did with `engine`.

Reply:
sure I will file a feature request

Reply:
Awesome!

Reply:
https://github.com/SeldonIO/seldon-core/issues/1559